tests/test_pretty.py

```diff
@@ -131,6 +131,75 @@ def test_recursive():
     assert result == expected
 
 
+def test_max_depth():
+    d = d = {}
+    d["foo"] = {"fob": {"a": [1, 2, 3], "b": {"z": "x", "y": ["a", "b", "c"]}}}
+
+    assert pretty_repr(d, max_depth=0) == "..."
```

Comment:
I'd probably check the case `max_depth=None` here.

Reply:
Good point